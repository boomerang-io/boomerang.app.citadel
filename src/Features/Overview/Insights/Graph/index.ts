// @ts-expect-error ts-migrate(6142) FIXME: Module './Graph' was resolved to 'C:/Users/Isabela... Remove this comment to see the full error message
export { default } from "./Graph"