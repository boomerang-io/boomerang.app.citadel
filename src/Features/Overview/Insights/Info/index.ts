// @ts-expect-error ts-migrate(6142) FIXME: Module './Info' was resolved to 'C:/Users/IsabelaM... Remove this comment to see the full error message
export { default } from "./Info"