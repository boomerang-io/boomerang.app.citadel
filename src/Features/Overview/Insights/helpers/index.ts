// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './helpers' or its correspondin... Remove this comment to see the full error message
export { default } from "./helpers"