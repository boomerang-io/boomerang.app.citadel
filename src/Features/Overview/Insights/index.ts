// @ts-expect-error ts-migrate(6142) FIXME: Module './Insights' was resolved to 'C:/Users/Isab... Remove this comment to see the full error message
export { default } from "./Insights"