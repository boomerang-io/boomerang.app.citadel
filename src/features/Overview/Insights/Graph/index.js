import React, { Component } from "react";
import PropTypes from "prop-types";
import { Tile } from "carbon-components-react";
import LineChart from "./LineChart";
import styles from "./graph.module.scss";

export class Graph extends Component {
  static propTypes = {
    formatedData: PropTypes.object
  };

  state = {
    fromDate: {},
    standard: {},
    toDate: {}
  };

  render() {
    const { formatedData } = this.props;

    return (
      <Tile
        style={{
          width: "100%",
          display: "flex",
          flexDirection: "column",
          alignItems: "spaceBetween",
          minHeight: "10rem"
        }}
      >
        <div className={styles.header}>
          <h3 className={styles.title}>Violations Trend</h3>
          <div className={styles.filters}></div>
        </div>
        <div className={styles.container} data-testid="insights-graph">
          {formatedData.chartData.length > 0 ? (
            <LineChart
              chartData={formatedData.chartData}
              lines={formatedData.lines}
              higherValue={formatedData.higherValue}
            />
          ) : (
            <p>No violations data to be found</p>
          )}
        </div>
      </Tile>
    );
  }
}

export default Graph;
