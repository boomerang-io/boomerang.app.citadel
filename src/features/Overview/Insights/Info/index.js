import React, { Component } from "react";
import PropTypes from "prop-types";
import { Tile } from "carbon-components-react";
import styles from "./info.module.scss";

export class Info extends Component {
  static propTypes = {
    info: PropTypes.object
  };

  render() {
    const { info } = this.props;
    return (
      <Tile
        style={{
          display: "flex",
          width: "100%",
          height: "10rem",
          alignItems: "center",
          padding: "0"
        }}
        data-testid="tile-info"
      >
        <div className={styles.counter} style={{ color: info.type === "gates" ? "#047cc0" : "#ffaa9d" }}>
          {info.count}
        </div>
        <div className={styles.divider}  />
        <div className={styles.text} >
          <h3 className={styles.title} >{info.title}</h3>
          <p className={styles.content} >{info.content}</p>
        </div>
      </Tile>
    );
  }
}

export default Info;
