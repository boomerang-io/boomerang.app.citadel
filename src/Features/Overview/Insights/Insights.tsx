import React from "react";
import Graph from "./Graph";
import Info from "./Info";
import { getLineChartData } from "./helpers/formatLineChartData";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './insights.module.scss' or its... Remove this comment to see the full error message
import styles from "./insights.module.scss";

type Props = {
    insights?: any[];
    policies?: any[];
    violations?: any[];
};

function Insights({ insights, violations, policies }: Props) {
  const formatedData = getLineChartData(insights);
  let gatesCount = 0;

  const gatesData = {
    type: "gates",
    title: "Current Gates",
    content: "Software never is perfect and it wonâ€™t be. But is that a license to create garbage?",
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    count: policies.reduce((acc, policy) => (gatesCount += policy.stages?.length ?? 0), 0),
  };
  const violationsData = {
    type: "violations",
    title: "Current Violations",
    content: "Code will set you free. Think of this as an opportunity for better code. Drive towards zero.",
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    count: violations.reduce((acc, violation) => (acc += violation?.nbrViolations ?? 0), 0),
  };

  return (
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    <section className={styles.container}>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <h2 className={styles.title}>Insights</h2>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <div className={styles.contentContainer}>
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <div className={styles.tilesContainer}>
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <Info info={gatesData} />
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <Info info={violationsData} />
        </div>
        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
        <div className={styles.graphsContainer}>
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <Graph formatedData={formatedData} />
        </div>
      </div>
    </section>
  );
}

export default Insights;
