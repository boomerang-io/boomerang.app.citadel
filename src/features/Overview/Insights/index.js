import React, { Component } from "react";
import PropTypes from "prop-types";
// import axios from "axios";
// import { connect } from "react-redux";
// import { bindActionCreators } from "redux";
// import { actions } from "State/teamMembers";
import Graph from "./Graph";
import Info from "./Info";
// import LoadingAnimation from "Components/Loading";
// import { notify, Notification } from "@boomerang/boomerang-components/lib/Notifications";
import { getLineChartData } from "./helpers/formatLineChartData";
import styles from "./insights.module.scss";

export class Insights extends Component {
  static propTypes = {
    insights: PropTypes.array,
    policies: PropTypes.array,
    violations: PropTypes.array
  };

  render() {
    const { insights, violations, policies } = this.props;
    const formatedData = getLineChartData(insights);
    let gatesCount = 0;
    policies.forEach(policy => gatesCount += policy.stages.length);

    const gatesData = {
      type: "gates",
      title: "Current Gates",
      content: "Software never is perfect and it wonâ€™t be. But is that a license to create garbage?",
      count: gatesCount
    };
    const violationsData = {
      type: "violations",
      title: "Current Violations",
      content: "Code will set you free. Think of this as an opportunity for better code. Drive towards zero.",
      count: violations.reduce((acc, violation) => (acc += violation.violations), 0)
    };

    return (
      <section className={styles.container}>
        <h2 className={styles.title}>Insights</h2>
        <div className={styles.contentContainer}>
          <div className={styles.tilesContainer}>
            <Info info={gatesData} />
            <Info info={violationsData} />
          </div>
          <div className={styles.graphsContainer}>
            <Graph formatedData={formatedData} />
          </div>
        </div>
      </section>
    );
  }
}

export default Insights;
