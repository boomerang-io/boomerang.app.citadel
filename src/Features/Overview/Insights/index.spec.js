import React from "react";
import Insights from "./index";

const insights = [
  {
    policyId: "5cddc15ff6ea74a9bbb2291b",
    policyName: "Marcus's Policy",
    ciPolicyCreatedDate: 1557791295946,
    insights: [
      {
        ciPolicyActivityId: "5cddc13fbc6f9400013360a9",
        policyActivityCreatedDate: 1557964800000,
        violations: 1
      },
      {
        ciPolicyActivityId: "5cddc177bc6f9400013360aa",
        policyActivityCreatedDate: 1557964800000,
        violations: 1
      },
      {
        ciPolicyActivityId: "5cddc430bc6f9400013360b2",
        policyActivityCreatedDate: 1557964800000,
        violations: 1
      },
      {
        ciPolicyActivityId: "5cddc44fbc6f9400013360b4",
        policyActivityCreatedDate: 1557964800000,
        violations: 1
      },
      {
        ciPolicyActivityId: "5ce1d56d76a4e100013eaab4",
        policyActivityCreatedDate: 1558304109636,
        violations: 1
      }
    ]
  }
];
const violations = [
  {
    policyId: "5cd49adff6ea74a9bb6adef4",
    policyName: "Tyson's Policy",
    ciStageId: "5cb123bdf6ea74a9bb5425b6",
    ciStageName: "dev",
    ciComponentId: "5cdb4c15f6ea74a9bb9cfdba",
    ciComponentName: "boomerang.service.ci",
    ciComponentVersionId: "5cdb4ca3f6ea74a9bb9cf251",
    ciComponentVersionName: "6.4.9-175",
    violations: 3
  }
];
const policies = [
  {
    createdDate: 1558051200000,
    name: "Timss",
    teamId: "5a8b331e262a70306622df73",
    definitions: [
      {
        policyTemplateId: "5cd49777f6ea74a9bb6ac629",
        rules: [
          {
            value: "sdfgsdfg",
            operator: "less than",
            metric: "complexity"
          }
        ]
      },
      {
        policyTemplateId: "5cd498f3f6ea74a9bb6ad0f3",
        rules: [
          {
            version: " asdfasdf",
            artifact: "asdasdf`",
            type: "maven"
          }
        ]
      }
    ],
    stages: ["dev"],
    id: "R4nuqJE"
  }
];

describe("Insights --- Snapshot", () => {
  it("Capturing Snapshot of Insights", async () => {
    const { baseElement } = global.rtlRender(<Insights insights={insights} policies={policies} violations={violations} />);
    expect(baseElement).toMatchSnapshot();
  });
});
