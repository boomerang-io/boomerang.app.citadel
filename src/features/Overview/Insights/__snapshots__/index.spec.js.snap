// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insights --- Snapshot Capturing Snapshot of Insights 1`] = `
<section
  className="container"
>
  <h2
    className="title"
  >
    Insights
  </h2>
  <div
    className="contentContainer"
  >
    <div
      className="tilesContainer"
    >
      <div
        className="bx--tile"
        data-testid="tile-info"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": "10rem",
            "padding": "0",
            "width": "100%",
          }
        }
      >
        <div
          className="counter"
          style={
            Object {
              "color": "#047cc0",
            }
          }
        >
          1
        </div>
        <div
          className="divider"
        />
        <div
          className="text"
        >
          <h3
            className="title"
          >
            Current Gates
          </h3>
          <p
            className="content"
          >
            Software never is perfect and it wonâ€™t be. But is that a license to create garbage?
          </p>
        </div>
      </div>
      <div
        className="bx--tile"
        data-testid="tile-info"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": "10rem",
            "padding": "0",
            "width": "100%",
          }
        }
      >
        <div
          className="counter"
          style={
            Object {
              "color": "#ffaa9d",
            }
          }
        >
          3
        </div>
        <div
          className="divider"
        />
        <div
          className="text"
        >
          <h3
            className="title"
          >
            Current Violations
          </h3>
          <p
            className="content"
          >
            Code will set you free. Think of this as an opportunity for better code. Drive towards zero.
          </p>
        </div>
      </div>
    </div>
    <div
      className="graphsContainer"
    >
      <div
        className="bx--tile"
        style={
          Object {
            "alignItems": "spaceBetween",
            "display": "flex",
            "flexDirection": "column",
            "minHeight": "10rem",
            "width": "100%",
          }
        }
      >
        <div
          className="header"
        >
          <h3
            className="title"
          >
            Violations Trend
          </h3>
          <div
            className="filters"
          />
        </div>
        <div
          className="container"
          data-testid="insights-graph"
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "width": "100%",
              }
            }
          >
            <div
              className="recharts-responsive-container"
              style={
                Object {
                  "height": "auto",
                  "maxHeight": undefined,
                  "minHeight": 100,
                  "minWidth": 0,
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "none",
                    "height": 0,
                    "position": "absolute",
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
